woocommerce/grow
# Upgrade JS package `generator-grow` to use Node.js v20

### Changes proposed in this Pull Request:

Part of #108

This PR:

- Bump dependencies and regenerate package-lock.json with Node.js v20.
- Set minimum support for Node.js to `"^18.17.0 || >=20.5.0"`, which align with the dependencies' minimum (`yeoman-generator@7.1.1`).
- Add the 'node:' prefix to explicitly indicate it requires a Node.js core module.
- Remove an unused file - **./generators/github/package.json**
- Fix the incompatible paths when composing generators.
- Fix the tests.

### Detailed test instructions:

1. Use Node.js >= 20.5.0
2. `cd packages/js/generator-grow`
3. `npm install`
4. `npm test` to see if tests can pass
5. `npm install -g yo`
6. `npm link`
7. Find a temporary empty directory
8. `npm init`
9. `yo grow` or `yo grow:github` to see if it can add boilerplate
10. Repeat the test with Node.js matches ^18.17.0
